File: fronter/src/model/assessment.ts
export interface Assessment {
  id: number;
  task_id: number;
  user_id: number;
  effectiveness_score: number;
  effort_score: number;
  impact_score: number;
  qualitative_feedback: string;
  created_at: string;
  updated_at: string;
}

export interface AddAssessment {
  task_id: number;
  user_id: number;
  effectiveness_score: number;
  effort_score: number;
  impact_score: number;
  qualitative_feedback: string;
}

File: backer/godotask/model/Assessment.go
package model

import "time"

type Assessment struct {
	ID                  int       `gorm:"primaryKey" json:"id"`
	TaskID              int       `json:"task_id"`
	UserID              int       `json:"user_id"`
	// 有効性スコア
	EffectivenessScore  int       `json:"effectiveness_score"`
	// 努力スコア
	EffortScore         int       `json:"effort_score"`
	// 印象値影響スコア
	ImpactScore         int       `json:"impact_score"`
	// 定性的フィードバック
	QualitativeFeedback string    `json:"qualitative_feedback"`
	CreatedAt           time.Time `json:"created_at"`
	UpdatedAt           time.Time `json:"updated_at"`
}

func (Assessment) TableName() string {
  return "assessments"
}
